common/fee/models.go

```diff
@@ -0,0 +1,183 @@
+// The baseline price is the maximum of the previous fee price attempt and the node's current fee price.
+func bumpFeePrice(cfg feetypes.BumpConfig, lggr logger.SugaredLogger, currentfeePrice, originalfeePrice, maxFeePriceInput *big.Int) (*big.Int, error) {
+	maxFeePrice := getMaxFeePrice(maxFeePriceInput, cfg.PriceMax()) // Make a wrapper config
+	bumpedFeePrice := bumpFeePriceByPercentage(originalfeePrice, cfg.BumpPercent(), cfg.BumpMin())
+
+	// Update bumpedFeePrice if currentfeePrice is higher than bumpedFeePrice and within maxFeePrice
+	bumpedFeePrice = maxBumpedFee(lggr, currentfeePrice, bumpedFeePrice, maxFeePrice, "fee price")
+
+	if bumpedFeePrice.Cmp(maxFeePrice) > 0 {
+		return maxFeePrice, errors.Wrapf(ErrBumpFeeExceedsLimit, "bumped fee price of %s would exceed configured max fee price of %s (original price was %s). %s",
+			bumpedFeePrice.String(), maxFeePrice, originalfeePrice.String(), label.NodeConnectivityProblemWarning)
+	} else if bumpedFeePrice.Cmp(originalfeePrice) == 0 {
+		// NOTE: This really shouldn't happen since we enforce minimums for
+		// FeeEstimator.BumpPercent and FeeEstimator.BumpMin in the config validation,
+		// but it's here anyway for a "belts and braces" approach
+		return bumpedFeePrice, errors.Wrapf(ErrBump, "bumped fee price of %s fee unit is equal to original fee price of %s fee unit."+
+			" ACTION REQUIRED: This is a configuration error, you must increase either "+
+			"FeeEstimator.BumpPercent or FeeEstimator.BumpMin", bumpedFeePrice.String(), originalfeePrice.String())
+	}
+	return bumpedFeePrice, nil
+}
+
+func getMaxFeePrice(userSpecifiedMax, maxFeePrice *big.Int) *big.Int {
+	return FeePriceLimit(userSpecifiedMax, maxFeePrice)
+}
+
+func bumpFeePriceByPercentage(originalFeePrice *big.Int, feeBumpPercent uint16, feeBumpUnits *big.Int) *big.Int {
+	return max(new(big.Int).Add(originalFeePrice, feeBumpUnits), addPercentage(originalFeePrice, feeBumpPercent))
+}
+
+func maxBumpedFee(lggr logger.SugaredLogger, currentFeePrice, bumpedFeePrice, maxFeePrice *big.Int, feeType string) *big.Int {
+	if currentFeePrice != nil {
+		if currentFeePrice.Cmp(maxFeePrice) > 0 {
+			// Shouldn't happen because the estimator should not be allowed to
+			// estimate a higher fee than the maximum allowed
+			lggr.AssumptionViolationf("Ignoring current %s of %s that would exceed max %s of %s", feeType, currentFeePrice.String(), feeType, maxFeePrice.String())
+		} else if bumpedFeePrice.Cmp(currentFeePrice) < 0 {
+			// If the current fee price is higher than the old price bumped, use that instead
+			bumpedFeePrice = currentFeePrice
+		}
+	}
+	return bumpedFeePrice
+}
+
+// CalculateFee computes the fee price and chain specific fee limit for a transaction.
+func CalculateFee(cfg feetypes.FixedPriceEstimatorConfig, bumpCfg feetypes.BumpConfig, feeLimit uint32, maxFeePrice *big.Int) (feePrice *big.Int, chainSpecificFeeLimit uint32, err error) {
```

Comment:
I initially thought that the config would be chain agnostic. However looking back now i think that passing values around  would be much simpler and more composable so that the other chains would not have to implement its own interface. Thanks for the note!